### PR TITLE
🌱 Add E2E test for KCP remediation

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR adds an E2E test for KCP remediation

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/3913
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/3656

### CODE REVIEW

File: test/e2e/mhc_remediations.go [hunk 1]
```diff
@@ -88,7 +88,40 @@ func MachineRemediationSpec(ctx context.Context, inputGetter func() MachineRemed
...
+		clusterResources = clusterctl.ApplyClusterTemplateAndWait(ctx, clusterctl.ApplyClusterTemplateAndWaitInput{
+			ClusterProxy: input.BootstrapClusterProxy,
+			ConfigCluster: clusterctl.ConfigClusterInput{
+				LogFolder:                filepath.Join(input.ArtifactFolder, "clusters", input.BootstrapClusterProxy.GetName()),
+				ClusterctlConfigPath:     input.ClusterctlConfigPath,
+				KubeconfigPath:           input.BootstrapClusterProxy.GetKubeconfigPath(),
+				InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+				Flavor:                   "kcp-remediation",
+				Namespace:                namespace.Name,
+				ClusterName:              fmt.Sprintf("%s-%s", specName, util.RandomString(6)),
```
Comment (sedefsavas) [inline test/e2e/mhc_remediations.go:114]: nit. We can add kcp to the cluster name to differentiate the logs.

File: test/e2e/data/infrastructure-docker/v1alpha4/cluster-template-md-remediation/md.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: cluster.x-k8s.io/v1alpha4
+kind: MachineDeployment
+metadata:
+  name: "${CLUSTER_NAME}-md-0"
+spec:
+  template:
+    metadata:
+      labels:
+        "e2e.remediation.label": ""
```
Comment (sedefsavas) [inline test/e2e/data/infrastructure-docker/v1alpha4/cluster-template-md-remediation/md.yaml:9]:
```suggestion
        "e2e.remediation.label": ""
        
```

**Review by vincepri — APPROVED** [review]
/approve
/milestone v0.4.0

**Pull request merged**